4.0 Limitations of security (#637)
neo4j/neo4j-documentation: cypher/cypher-docs/src/docs/dev/ql/administration/security/security-and-db-operations.asciidoc

> The rules of a security model can have some impact on some of the database operations.

Reviewer: ```suggestion
The rules of a security model may impact some of the database operations.
```

This comes down to necessary additional security checks that incur additional data accesses.

> Especially when it comes to count store operations, as they are usually very fast lookups, this difference might be noticeable.

Reviewer: ```suggestion
Especially in regards to count store operations, as they are usually very fast lookups, the difference might be noticeable.
```

> Lets look at the following security rules that setup a "restricted" and a "free" role as an example:

Reviewer: ```suggestion
Let's look at the following security rules that set up a "restricted" and a "free" role as an example:
```

> Lets look at the following security rules that setup a "restricted" and a "free" role as an example:

Reviewer 2: Maybe the role names "restricted" and "free" should be backticked instead of in quotes? Same in the table header below

     GRANT TRAVERSE ON GRAPH * NODES Person TO restricted;
     DENY TRAVERSE ON GRAPH * NODES Customer TO restricted;
     GRANT TRAVERSE ON GRAPH * ELEMENTS * TO free;

> Now, lets look at what the database needs to do in order to execute the following query:

Reviewer: ```suggestion
Now, let's look at what the database needs to do in order to execute the following query:
```

     MATCH (n:Person) RETURN count(n);

For both roles the execution plan will look like this:

[listing]
....
+--------------------------+
| Operator                 |
+--------------------------+
| +ProduceResults          |
| |                        +
| +NodeCountFromCountStore |
+--------------------------+
....

> Internally however, very different operations need to be executed. The following table illustrate the difference.

Reviewer: ```suggestion
Internally however, very different operations need to be executed. The following table illustrates the difference.
```

[%header,cols=2*]
|===
|User with "free" role
|User with "restricted" role

|The database can access the count store and retrieve the total number of nodes with the label `:Person`.

This is a very quick operation.

> |The database can not just access the count store because it must make sure that only nodes with the desired label `:Person` are counted.

Reviewer: ```suggestion
|The database cannot just access the count store because it must make sure that only nodes with the desired label `:Person` are counted.
```

> |The database can not just access the count store because it must make sure that only nodes with the desired label `:Person` are counted.

Reviewer 2: Maybe `only traversable nodes` or something like that?

> Do to this, each node with the `:Person` label needs to be accessed and examined to make sure that it does not also have any blacklisted label like `:Customer`.

Reviewer: ```suggestion
Due to this, each node with the `:Person` label needs to be accessed and examined to make sure that it does not also have a denylisted label, such as `:Customer`.
```
Also, where was the label `:Customer` denylisted (blacklisted, see other comment?

Reviewer 3: Line 8:
DENY TRAVERSE ON GRAPH * NODES Customer TO restricted;
